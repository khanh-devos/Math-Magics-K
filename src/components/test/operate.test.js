import operate from '../logic/operate';

describe('test operate', () => {
    const [a, b] = [10, 0];
    const arr = ['+', '-', 'x', 'รท', '%'];
    arr.forEach(e => {
        it(`use snaphsop ${a} ${e} ${b} = ?`, () => {
            expect(operate(a, b, e)).toMatchSnapshot();
        })  
    })

    it(`use snaphsop ${a} ${'#'} ${b} = ?`, () => {
        expect(() => operate(a, b, '#')).toThrow("Unknown operation '#'");
    })
})
