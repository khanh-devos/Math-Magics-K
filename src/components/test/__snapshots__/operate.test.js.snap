// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test operate use snaphsop 10 % 0 = ? 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`test operate use snaphsop 10 + 0 = ? 1`] = `"10"`;

exports[`test operate use snaphsop 10 - 0 = ? 1`] = `"10"`;

exports[`test operate use snaphsop 10 x 0 = ? 1`] = `"0"`;

exports[`test operate use snaphsop 10 รท 0 = ? 1`] = `"Can't divide by 0."`;
